### PR TITLE
bpo-35610: IDLE - Replace .context_use_ps1 with .prompt_last_line

### PR DESCRIPTION
**Pull request by csabella**
#4143 for issue 31858 fixed some uses of the shell prompt and removed setting `context_use_ps1` in pyshell, but not in editor.  That partial removal was a mistake since the attribute was still used in hyperparser (and some tests).  #11346 for issue 34055 reversed the removal. This PR removes both settings of 'context_use_ps1' and replaces all uses with 'prompt_last_line, which is only non-empty in Shell.

https://bugs.python.org/issue35610

### CODE REVIEW

**Review by auvipy — APPROVED** [review]

**Comment by terryjreedy** [comment]
@auvipy When you review, please leave a comment that might help me make a decision.  I don't know what mere '[x] Approve' means beyond the CI checks passing.

File: Misc/NEWS.d/next/IDLE/2018-12-24-17-17-21.bpo-34055.zDC3Va.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+IDLE Shell: Fix printing smart indent on every line
```
Comment (terryjreedy) [inline Misc/NEWS.d/next/IDLE/2018-12-24-17-17-21.bpo-34055.zDC3Va.rst]: This is a holdover from when this PR was written for issue 34055.  The fix was made separately there.  This patch only cleans the code and should have no visible behavior change.  I will change this when I add an idlelib.NEWS entry.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
I will make the changes requested and add a couple of comments.

**Review by terryjreedy — APPROVED** [review]
I verified that grepping idlelib recursively for 'context_use_ps1' returns nothing.
Will merge.

**Pull request merged**